### PR TITLE
Replace frame mapping text by ref. to tp draft.

### LINKED ISSUE #4097: TLS spec should not specify mapping of frames to packets, except for CRYPTO
There is a long development in the section 4 of the TLS draft to explain:
```
Some frames are prohibited in different packet number spaces.  The
   rules here generalize those of TLS, in that frames associated with
   establishing the connection can usually appear in packets in any
   packet number space, whereas those associated with transferring data
   can only appear in the application data packet number space:

   *  PADDING, PING, and CRYPTO frames MAY appear in any packet number
      space.

   *  CONNECTION_CLOSE frames signaling errors at the QUIC layer (type
      0x1c) MAY appear in any packet number space.  CONNECTION_CLOSE
      frames signaling application errors (type 0x1d) MUST only appear
      in the application data packet number space.

   *  ACK frames MAY appear in any packet number space, but can only
      acknowledge packets that appeared in that packet number space.
      However, as noted below, 0-RTT packets cannot contain ACK frames.

   *  All other frame types MUST only be sent in the application data
      packet number space.

   Note that it is not possible to send the following frames in 0-RTT
   packets for various reasons: ACK, CRYPTO, HANDSHAKE_DONE, NEW_TOKEN,
   PATH_RESPONSE, and RETIRE_CONNECTION_ID.  A server MAY treat receipt
   of these frames in 0-RTT packets as a connection error of type
   PROTOCOL_VIOLATION.
```
I think that this text does not belong in the specification of **TLS**, which should stick to just explaining how to send the CRYPTO frames. Could it be replaced by a reference to the transport draft?

### PR DESCRIPTION
**Pull request by huitema**
Close #4097

### CODE REVIEW

**Review by martinthomson — CHANGES_REQUESTED** [review]
This isn't quite enough as the transport draft contains reference to this section of the document.  I would prefer that this text be moved under a new Section 12.5 and the text "Section 4 of [QUIC-TLS] provides more detail about these restrictions." in [Section 12.4](https://quicwg.org/base-drafts/draft-ietf-quic-transport.html#section-12.4-10) adjusted to point to that.  There might be other citations for Section 4 of [QUIC-TLS] as well that would need adjustment.

File: draft-ietf-quic-tls.md [hunk 1]
```diff
@@ -325,32 +325,8 @@ One important difference between TLS records (used with TCP) and QUIC CRYPTO
...
-
-- All other frame types MUST only be sent in the application data packet number
-  space.
-
-Note that it is not possible to send the following frames in 0-RTT packets for
-various reasons: ACK, CRYPTO, HANDSHAKE_DONE, NEW_TOKEN, PATH_RESPONSE, and
-RETIRE_CONNECTION_ID.  A server MAY treat receipt of these frames in 0-RTT
-packets as a connection error of type PROTOCOL_VIOLATION.
+into the same packet. Some frames are prohibited in different packet number
+spaces, as specified in section 12.4 of {{QUIC-TRANSPORT}}.
```
[the remark this answers is not in the document]
Comment (huitema) [reply draft-ietf-quic-tls.md]: Just pushed a commit to do that.

File: draft-ietf-quic-transport.md [hunk 2]
```diff
@@ -3487,6 +3487,33 @@ This rule applies to all current and future QUIC frame types.  An endpoint
 MAY treat the receipt of a frame type that uses a longer encoding than
 necessary as a connection error of type PROTOCOL_VIOLATION.
 
+## Frames and Number Spaces {#frames_and_spaces}
```
Comment (martinthomson) [inline draft-ietf-quic-transport.md:3490]: We use hyphens mostly.

```suggestion
## Frames and Number Spaces {#frames-and-spaces}
```

And up on line 3443 (where I can't comment) replace: "Section 4 of {{QUIC-TLS}}" with "{{frames-and-spaces}}".

**Review by martinthomson — COMMENTED** [review]
Thanks for moving it.  It's easier that way.

There is another reference to Section 4 of the TLS spec in Section 19.19 as well. (That's the bottom of CONNECTION_CLOSE.)

File: draft-ietf-quic-transport.md (hunk 2 above)
Comment (huitema) [reply draft-ietf-quic-transport.md:3490]: Give me couple hours, I will fix that after dinner.

**Comment by huitema** [comment]
There were two references, I fixed them both.

**Review by martinthomson — APPROVED** [review]
Thanks Christian, this looks good.  I'll give @janaiyengar a chance to review also, but I think it's worth taking this.

**Review by ianswett — APPROVED** [review]

File: draft-ietf-quic-tls.md [hunk 3]
```diff
@@ -325,32 +325,8 @@ One important difference between TLS records (used with TCP) and QUIC CRYPTO
...
-
-- All other frame types MUST only be sent in the application data packet number
-  space.
-
-Note that it is not possible to send the following frames in 0-RTT packets for
-various reasons: ACK, CRYPTO, HANDSHAKE_DONE, NEW_TOKEN, PATH_RESPONSE, and
-RETIRE_CONNECTION_ID.  A server MAY treat receipt of these frames in 0-RTT
-packets as a connection error of type PROTOCOL_VIOLATION.
+into the same packet. Some frames are prohibited in different packet number
+spaces, as specified in section 12.5 of {{QUIC-TRANSPORT}}.
```
Comment (janaiyengar) [inline draft-ietf-quic-tls.md:329]:
```suggestion
spaces; see section 12.5 of {{QUIC-TRANSPORT}}.
```

File: draft-ietf-quic-transport.md [hunk 4]
```diff
@@ -3487,6 +3487,33 @@ This rule applies to all current and future QUIC frame types.  An endpoint
 MAY treat the receipt of a frame type that uses a longer encoding than
 necessary as a connection error of type PROTOCOL_VIOLATION.
 
+## Frames and Number Spaces {#frames-and-spaces}
```
Comment (janaiyengar) [inline draft-ietf-quic-transport.md:3490]:
```suggestion
## Frames and Packet Number Spaces {#frames-and-spaces}
```

**Review by janaiyengar — APPROVED** [review]
Makes sense, thanks for writing this up, @huitema.

File: draft-ietf-quic-transport.md (hunk 4 above)
Comment (huitema) [reply draft-ietf-quic-transport.md:3490]: Done.

**Pull request merged**